### PR TITLE
Voting rate limit

### PR DESCRIPTION
**Pull request by jimrandomh**
Adds rate limits to voting: 50 per 24 hours, 20 per 1 hour, and 10 on the same user per 24 hours.

Lightly tested, mainly just the plumbing: I lowered the rate limits, did some votes on my local install to get up to the limit, and confirmed that rate-limit errors started appearing and were presented reasonably.

### CODE REVIEW

**Comment by jimrandomh** [comment]
Lint issue was not introduced in this PR; it was introduced in https://github.com/LessWrong2/Lesswrong2/pull/1749 .

**Comment by Raemon** [comment]
I've looked over the code and seems to make sense.

Just want to doublecheck that you've tested the cases (setting the vote limits low if need be)

1. for total votes per day
2. for total votes per hour (maybe this is similar enough to per-day that doesn't need to be separately tested)
3. for total votes on an author per day (but that you can still vote on other authors)
4. that if you log in as another user you're not still facing any voting limits (after having run into one of the above limits)
5. that cancelled votes are properly not-counting.

Assuming you have confidence those cases are working, and merge off the new devel to fix lint issues, I'd be fine approving.

**Comment by jimrandomh** [comment]
I haven't tested it that thoroughly, and... that level of testing is too much of a pain to feel worth it.

File: packages/lesswrong/server/voteServer.js [hunk 1]
```diff
@@ -229,5 +214,43 @@ export const performVoteServer = async ({ documentId, document, voteType = 'bigU
   // const newDocument = collection.findOne(documentId);
   document.__typename = collection.options.typeName;
   return document;
+}
+
+const getVotingRateLimits = async (user) => {
+  return {
+    perDay: 50,
```
Comment (Discordius) [inline packages/lesswrong/server/voteServer.js]:
```suggestion
    perDay: 100,
```

File: packages/lesswrong/server/voteServer.js [hunk 2]
```diff
@@ -229,5 +214,43 @@ export const performVoteServer = async ({ documentId, document, voteType = 'bigU
...
+    perHour: 20,
```
Comment (Discordius) [inline packages/lesswrong/server/voteServer.js]:
```suggestion
    perHour: 30,
```

File: packages/lesswrong/server/voteServer.js [hunk 3]
```diff
@@ -229,5 +214,43 @@ export const performVoteServer = async ({ documentId, document, voteType = 'bigU
...
+    perUserPerDay: 10,
```
Comment (Discordius) [inline packages/lesswrong/server/voteServer.js]: 30

Comment (Discordius) [inline packages/lesswrong/server/voteServer.js]:
```suggestion
    perUserPerDay: 30,
```

**Comment by Raemon** [comment]
Can you simulate it in your head? (part of the point is not necessarily to do all the testing but to have thought through ramifications that you might not have thought of, systematically)

**Comment by jimrandomh** [comment]
Because of the time-based component of this code, testing cases individually would require either waiting a long time between tests, or fiddly manual database operations, neither of which are worth it. I'm fairly confident that this code is correct.

**Pull request merged**